src/jdataview.js

```diff
@@ -480,354 +213,199 @@ var proto = jDataView.prototype = {
-	},
-
-	_setBinaryFloat: function (byteOffset, value, mantSize, expSize, littleEndian) {
-		var signBit = value < 0 ? 1 : 0,
-			exponent,
-			mantissa,
-			eMax = ~(-1 << (expSize - 1)),
-			eMin = 1 - eMax;
-
-		if (value < 0) {
-			value = -value;
-		}
+	}
 
-		if (value === 0) {
-			exponent = 0;
-			mantissa = 0;
-		} else if (isNaN(value)) {
-			exponent = 2 * eMax + 1;
-			mantissa = 1;
-		} else if (value === Infinity) {
-			exponent = 2 * eMax + 1;
-			mantissa = 0;
-		} else {
-			exponent = Math.floor(Math.log(value) / Math.LN2);
-			if (exponent >= eMin && exponent <= eMax) {
-				mantissa = Math.floor((value * pow2(-exponent) - 1) * pow2(mantSize));
-				exponent += eMax;
-			} else {
-				mantissa = Math.floor(value / pow2(eMin - mantSize));
-				exponent = 0;
-			}
-		}
+	getSigned(bitLength, byteOffset) {
+		const shift = 32 - bitLength;
+		return this.getUnsigned(bitLength, byteOffset) << shift >> shift;
+	}
 
-		var b = [];
-		while (mantSize >= 8) {
-			b.push(mantissa % 256);
-			mantissa = Math.floor(mantissa / 256);
-			mantSize -= 8;
-		}
-		exponent = (exponent << mantSize) | mantissa;
-		expSize += mantSize;
-		while (expSize >= 8) {
-			b.push(exponent & 0xff);
-			exponent >>>= 8;
-			expSize -= 8;
-		}
-		b.push((signBit << expSize) | exponent);
+	getUnsigned(bitLength, byteOffset) {
+		const value =
+			this.#getBitRangeData(bitLength, byteOffset).wideValue >>>
+			-this._bitOffset;
+		return bitLength < 32 ? value & ~(-1 << bitLength) : value;
+	}
 
-		this._setBytes(byteOffset, b, littleEndian);
-	},
 
-	_setFloat32: function (byteOffset, value, littleEndian) {
-		this._setBinaryFloat(byteOffset, value, 23, 8, littleEndian);
-	},
+	setUnsigned(byteOffset, value, bitLength) {
+		const data = this.#getBitRangeData(bitLength, byteOffset);
+		const b = data.bytes;
+		let wideValue = data.wideValue;
 
-	_setFloat64: function (byteOffset, value, littleEndian) {
-		this._setBinaryFloat(byteOffset, value, 52, 11, littleEndian);
-	},
+		wideValue &= ~(~(-1 << bitLength) << -this._bitOffset); // clearing bit range before binary "or"
+		wideValue |=
+			(bitLength < 32 ? value & ~(-1 << bitLength) : value) << -this._bitOffset; // setting bits
 
-	_set64: function (Type, byteOffset, value, littleEndian) {
-		if (typeof value !== 'object') {
-			value = Type.fromNumber(value);
+		for (let i = b.length - 1; i >= 0; i--) {
+			b[i] = wideValue & 0xff;
+			wideValue >>>= 8;
 		}
 
-		littleEndian = defined(littleEndian, this._littleEndian);
-		byteOffset = defined(byteOffset, this._offset);
-
-		var parts = littleEndian ? {lo: 0, hi: 4} : {lo: 4, hi: 0};
+		this.#setBytes(data.start, b, true);
+	}
 
-		for (var partName in parts) {
-			this.setUint32(byteOffset + parts[partName], value[partName], littleEndian);
-		}
+	setSigned(byteOffset, value, bitLength) {
+		return this.setUnsigned(byteOffset, value, bitLength);
+	}
 
-		this._offset = byteOffset + 8;
-	},
-
-	setInt64: function (byteOffset, value, littleEndian) {
```

Comment:
Let's keep those methods. While BigInt is the way to go for large numbers nowadays, there are still lots of scenarios where it's convenient to allow writing numbers (up to 2^53) as if they were 64 bit.

Since new methods are called `setBigInt64`, there should be no name clash anyway.

Reply:
Alright I've added them. They just convert the input to a bigint or return value to a number and pass it off to the bigint methods.

When I was adding extra tests for the new methods, I removed the old compareInt64 helper function, which converted them back to regular numbers before comparing them. However, In doing so, one of the tests for `bigUint64` (https://github.com/zadeviggers/jDataView/blob/3df8b2b0f0bac9933dca152b1df7c417398b6c2f/test/getters.test.js#L210) started failing. Looking at it, it seems like the test was relying on loss of precision to pass. Should I just update it, or am I on the wrong track here?